sdk/node/test/accounts.js

```diff
@@ -0,0 +1,177 @@
+      })
+
+      it('account creation rejected due to missing key fields', () => {
+        return expect(client.accounts.create({alias: 'david'}))
+        .to.be.rejectedWith('CH202')
+      })
+    })
+
+    describe('Batch account creation', () => {
+      let batchResponse = {}
+
+      before(() => client.accounts.createBatch([
+          {alias: `carol-${uuid.v4()}`, rootXpubs: [mockHsmKey.xpub], quorum: 1}, // success
+          {alias: 'david'}, // failure
+          {alias: `eve-${uuid.v4()}`, rootXpubs: [mockHsmKey.xpub], quorum: 1}, // success
+        ])
+        .then(resp => {batchResponse = resp})
+      )
+
+      it('returns two successes', () => assert.equal(batchResponse.successes[1], null))
+      it('returns one error', () => assert.deepEqual([batchResponse.errors[0], batchResponse.errors[2]], [null, null]))
+    })
+
+    describe('Single account tags update', () => {
+
+      it('successfully updates account tags', () => {
+        return client.accounts.updateTags({
+          alias: xAccountAlias,
+          tags: {x: 1},
+        })
+        .then(() => {
+          return client.accounts.query({
+            filter: `alias='${xAccountAlias}'`
+          })
+        })
+        .then(page => {
+          assert.deepEqual(page.items[0].tags, {x: 1})
+        })
+      })
+
+      it('fails to update account tags', () => {
+        return expect(
+          client.accounts.updateTags({
+            // ID/Alias intentionally omitted
+            tags: {x: 1},
+          })
+        ).to.be.rejectedWith('CH051')
+      })
+    })
+
+    describe('Batch account tags update', () => {
+
+      it('successfully updates accounts tags', () => {
+        return client.accounts.updateTagsBatch([{
+          alias: xAccountAlias,
+          tags: {x: 2},
+        }, {
+          alias: yAccountAlias,
+          tags: {y: 2},
+        }])
+        .then(() => {
+          return client.accounts.query({
+            filter: `alias='${xAccountAlias}' OR alias='${yAccountAlias}'`
+          })
+        })
+        .then(page => {
+          assert.deepEqual(page.items.find(i => i.alias.match(/^x-/)).tags, {x: 2})
+          assert.deepEqual(page.items.find(i => i.alias.match(/^y-/)).tags, {y: 2})
+        })
+      })
+
+      it('fails to update accounts tags with missing id', () => {
+        return client.accounts.updateTagsBatch([{
+          alias: xAccountAlias,
+          tags: {x: 3},
+        }, {
+          // ID/Alias intentionally omitted
+          tags: {y: 3}
+        }])
+        .then(batch => {
+          assert(batch.successes[0])
+          assert(!batch.successes[1])
+          assert(!batch.errors[0])
+          assert(batch.errors[1])
+        })
+      })
+    })
+  })
+
+  // These just test that the callback is engaged correctly. Behavior is
+  // tested in the promises test.
+  describe('Callback style', () => {
+
+    it('Single account creation', (done) => {
+      client.accounts.create(
```

Comment:
If we are just trying to prove that the callback is being invoked at all, this is one alternative:

```
client.accounts.create(
  {}, // intentionally blank,
  done
)
```

Like I said in chat, I'm not sure which one I prefer...what do you think?